mock/src/main/java/com/telefonica/mock/MockApiClient.kt

```diff
@@ -11,14 +11,31 @@ import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 
 open class MockApiClient @Inject constructor(
-    private val dispatcher: CoroutineDispatcher,
+    private val coroutineDispatcher: CoroutineDispatcher,
     private val mockWebServer: MockWebServer,
 ) {
 
     private val mockMap: MutableMap<String, List<Mock>> = mutableMapOf()
 
+    private val enqueuedAnswers: MutableList<Mock> = mutableListOf()
+
+    private val dispatcher = object : Dispatcher() {
+        override fun dispatch(request: RecordedRequest): MockResponse {
+            val mockFiles: List<Mock> = enqueuedAnswers
+                .filter { mock -> request.method == mock.method?.value }
```

Comment:
Just curious, why method could be null?

Reply:
When we developed this part, it made sense, maybe we could set it to not null and not leave it as the default.